src/tasks/migrations/utils.js

```diff
@@ -0,0 +1,242 @@
+ * @param  {String} component name of component
+ * @return {Promise<Boolean>}
+ */
+const checkComponentExists = async (api, component) => {
+  try {
+    const componentData = await getComponentsFromName(api, component)
+
+    return Promise.resolve(Object.keys(componentData).length > 0)
+  } catch (e) {
+    const error = parseError(e)
+    return Promise.reject(error.error)
+  }
+}
+
+/**
+ * @method checkFileExists
+ * @param  {String} filePath
+ * @return {Promise<Boolean>}
+ */
+const checkFileExists = async (filePath) => fs.pathExists(filePath)
+
+/**
+ * @method createMigrationFile
+ * @param  {String} fileName path to file
+ * @param  {String} field    name of the field
+ * @return {Promise<Boolean>}
+ */
+const createMigrationFile = (fileName, field) => {
+  console.log(`${chalk.blue('-')} Creating the migration file in migrations folder`)
+
+  // use lodash.template to replace the occurrences of fieldname
+  const compile = template(migrationTemplate, {
+    interpolate: /{{([\s\S]+?)}}/g
+  })
+  const outputMigrationFile = compile({
+    fieldname: field
+  })
+
+  return fs.outputFile(getPathToFile(fileName), outputMigrationFile)
+}
+
+/**
+ * @method getInquirerOptions
+ * @param  {String} type
+ * @return {Array}
+ */
+const getInquirerOptions = (type) => {
+  if (type === 'file-exists') {
+    return [{
+      type: 'confirm',
+      name: 'choice',
+      message: 'Do you want to continue? (This will overwrite the content of the file!)'
+    }]
+  }
+
+  return []
+}
+
+/**
+ * @method showMigrationChanges
+ * @param  {String} path      field name
+ * @param  {unknown} value    updated value
+ * @param  {unknown} oldValue previous value
+ */
+const showMigrationChanges = (path, value, oldValue) => {
+  // It was created a new field
+  if (oldValue === undefined) {
+    // truncate the string with more than 30 characters
+    const _value = truncate(value)
+
+    console.log(`  ${chalk.green('-')} Created field "${chalk.green(path)}" with value "${chalk.green(_value)}"`)
+    return
+  }
+
+  // It was removed the field
+  if (value === undefined) {
+    console.log(`  ${chalk.red('-')} Removed the field "${chalk.red(path)}"`)
+    return
+  }
+
+  // It was updated the value
+  if (value !== oldValue) {
+    // truncate the string with more than 30 characters
+    const _value = truncate(value)
+    const _oldValue = truncate(oldValue)
+
+    console.log(`  ${chalk.blue('-')} Updated field "${chalk.blue(path)}" from "${chalk.blue(_oldValue)}" to "${chalk.blue(_value)}"`)
+  }
+}
+
+/**
+ * @method processMigration
+ * @param  {Object}   content component structure from Storyblok
+ * @param  {String}   component    name of the component that is processing
+ * @param  {Function} migrationFn  the migration function defined by user
+ * @param  {Boolean}  isDryrun     if true, watch changes
+ * @return {Promise<Boolean>}
+ */
+const processMigration = async (content = {}, component = '', migrationFn, isDryrun) => {
```

Comment:
Why async here? Was your idea to add "await" in front of migrationFn? I think that would be a good idea.

Reply:
Great! I'll use in front of migrationFn.